checker/jdk/nullness/src/org/w3c/dom/Attr.java

```diff
@@ -0,0 +1,307 @@
+ * changed, and it is of a different type than CDATA, it may normalize it
+ * again at that time. This is especially true of specialized DOM
+ * implementations, such as SVG DOM implementations, which store attribute
+ * values in an internal form different from a string.
+ * <p>The following table gives some examples of the relations between the
+ * attribute value in the original document (parsed attribute), the value as
+ * exposed in the DOM, and the serialization of the value:
+ * <table border='1' cellpadding='3'>
+ * <tr>
+ * <th>Examples</th>
+ * <th>Parsed
+ * attribute value</th>
+ * <th>Initial <code>Attr.value</code></th>
+ * <th>Serialized attribute value</th>
+ * </tr>
+ * <tr>
+ * <td valign='top' rowspan='1' colspan='1'>
+ * Character reference</td>
+ * <td valign='top' rowspan='1' colspan='1'>
+ * <pre>"x&amp;#178;=5"</pre>
+ * </td>
+ * <td valign='top' rowspan='1' colspan='1'>
+ * <pre>"x\u00b2=5"</pre>
+ * </td>
+ * <td valign='top' rowspan='1' colspan='1'>
+ * <pre>"x&amp;#178;=5"</pre>
+ * </td>
+ * </tr>
+ * <tr>
+ * <td valign='top' rowspan='1' colspan='1'>Built-in
+ * character entity</td>
+ * <td valign='top' rowspan='1' colspan='1'>
+ * <pre>"y&amp;lt;6"</pre>
+ * </td>
+ * <td valign='top' rowspan='1' colspan='1'>
+ * <pre>"y&lt;6"</pre>
+ * </td>
+ * <td valign='top' rowspan='1' colspan='1'>
+ * <pre>"y&amp;lt;6"</pre>
+ * </td>
+ * </tr>
+ * <tr>
+ * <td valign='top' rowspan='1' colspan='1'>Literal newline between</td>
+ * <td valign='top' rowspan='1' colspan='1'>
+ * <pre>
+ * "x=5&amp;#10;y=6"</pre>
+ * </td>
+ * <td valign='top' rowspan='1' colspan='1'>
+ * <pre>"x=5 y=6"</pre>
+ * </td>
+ * <td valign='top' rowspan='1' colspan='1'>
+ * <pre>"x=5&amp;#10;y=6"</pre>
+ * </td>
+ * </tr>
+ * <tr>
+ * <td valign='top' rowspan='1' colspan='1'>Normalized newline between</td>
+ * <td valign='top' rowspan='1' colspan='1'>
+ * <pre>"x=5
+ * y=6"</pre>
+ * </td>
+ * <td valign='top' rowspan='1' colspan='1'>
+ * <pre>"x=5 y=6"</pre>
+ * </td>
+ * <td valign='top' rowspan='1' colspan='1'>
+ * <pre>"x=5 y=6"</pre>
+ * </td>
+ * </tr>
+ * <tr>
+ * <td valign='top' rowspan='1' colspan='1'>Entity <code>e</code> with literal newline</td>
+ * <td valign='top' rowspan='1' colspan='1'>
+ * <pre>
+ * &lt;!ENTITY e '...&amp;#10;...'&gt; [...]&gt; "x=5&amp;e;y=6"</pre>
+ * </td>
+ * <td valign='top' rowspan='1' colspan='1'><em>Dependent on Implementation and Load Options</em></td>
+ * <td valign='top' rowspan='1' colspan='1'><em>Dependent on Implementation and Load/Save Options</em></td>
+ * </tr>
+ * </table>
+ * <p>See also the <a href='http://www.w3.org/TR/2004/REC-DOM-Level-3-Core-20040407'>Document Object Model (DOM) Level 3 Core Specification</a>.
+ */
+public interface Attr extends Node {
+    /**
+     * Returns the name of this attribute. If <code>Node.localName</code> is
+     * different from <code>null</code>, this attribute is a qualified name.
+     */
+    @Pure public String getName();
```

Comment:
Every attribute is attribute node. From this [reference](https://www.w3.org/2003/01/dom2-javadoc/org/w3c/dom/Node.html) first row in tabular data, the results of `getName()` and `getValue()` method are similar to `getNodeName()` and `getNodeValue()` from [Node.java](https://github.com/typetools/checker-framework/blob/master/checker/jdk/nullness/src/org/w3c/dom/Node.java) . This both entities `name` and `value` cannot have null value.